cpp/src/arrow/dataset/scanner.h

```diff
@@ -104,7 +143,7 @@ class ARROW_DS_EXPORT ScannerBuilder {
   ScannerBuilder* IncludePartitionKeys(bool include = true);
 
```

Comment:
Not in this PR, but the docstring for `IncludePartitionKeys` is incomplete ("otherwise"... what?).